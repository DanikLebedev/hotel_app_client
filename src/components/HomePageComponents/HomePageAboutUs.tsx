import React from 'react'
import {Col, Container, Row} from "react-bootstrap";
import {NavLink} from "react-router-dom";
import {Slider} from "../Slider/Slider";
import roomImage1 from "../../assets/images/vertical-city-hotel.jpg";
import roomImage2
    from "../../assets/images/38736452-luxurious-bed-with-cushion-in-royal-bedroom-interior-hotel-room-in-bright-colors-vertical-orientatio.jpg";


export const HomePageAboutUs = () => {
    return (

        <section className={'home-page__about-us'}>
            <Container className={'main__title'}>
                <h1 className='header__title'>Welcome to Rixos Hotel</h1>
                <p className={'header__subtitle'}>Chilling out on the bed in your hotel room watching TV, while wearing your own pajamas, is sometimes the best part of vacation.</p>
                <div className={"button__container"}>
                    <NavLink to={'/rooms'} >
                        <button className='button'>See rooms</button>
                    </NavLink>
                    <NavLink to={'/about'} >
                        <button className={'btn-transparent button'}>About us</button>
                    </NavLink>
                </div>
            </Container>
            <Slider/>
            <Container>
                <Row>
                    <Col lg={6} md={6}>
                        <h2 className={'section__title'}>Hotels to celebrate life</h2>
                        <p className={'home-page__about-us-text'}>We are proud to say that since our opening in '98
                            we have been serving our visitors in
                            the best possible way.
                            In Rixos Hotel, where each one of four 17 rooms - with its own personality and style -
                            will help you to feel at home and to live the 'Rixos' experience as you would never have
                            imagined it</p>
                        <p className={'home-page__about-us-text'}>Make memories at Rixos hotel, Amelia Island, where
                            southern charm, magnificent, scenery
                            and casually elegant surroundings exemplify the gentle ambience of his barrier island'
                            luxury beachfront resort</p>
                        <button className={'button btn-black'}>Read more</button>
                    </Col>
                    <Col lg={6} md={6}>
                        <div
                            className={'home-page__about-us-photos d-flex justify-content-center align-items-center'}>
                            <img src={roomImage1} alt="room"/>
                            <img src={roomImage2} alt="room"/>
                        </div>
                    </Col>
                </Row>
            </Container>
        </section>
    )
}