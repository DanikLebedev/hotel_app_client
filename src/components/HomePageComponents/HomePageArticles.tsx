import React from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import galleryPhoto1 from '../../assets/images/cream_candles_flowers_green_therapy_11044_1600x1200.jpg';
import galleryPhoto2 from '../../assets/images/thumb-1920-206717.jpg'
import galleryPhoto3 from '../../assets/images/10-103370_salad-food-wallpaper-food-images-high-resolution.jpg';
import galleryPhoto4 from '../../assets/images/high-resolution-gif-1.gif';

export const HomePageArticles = () => {
    return (
        <Container fluid={true}>
            <Row>
                <Col
                    lg={4}
                    md={3}
                    sm={3}
                    className={'home-page-articles-item'}
                    style={{ background: `url(${galleryPhoto1}) center center / cover`, height: '20rem' }}
                >
                    <div className={'article-info'}>
                        <h4>Health SPA</h4>
                        <p>Lorem ipsum dolor sit amet.</p>
                        <a href="/">Read more</a>
                    </div>
                </Col>
                <Col
                    lg={4}
                    md={3}
                    sm={3}
                    className={'home-page-articles-item'}
                    style={{ background: `url(${galleryPhoto2}) center center / cover`, height: '20rem' }}
                >
                    <div className={'article-info'}>
                        <h4>Golf</h4>
                        <p>Lorem ipsum dolor sit amet.</p>
                        <a href="/">Read more</a>
                    </div>
                </Col>
                <Col
                    lg={4}
                    md={3}
                    sm={3}
                    className={'home-page-articles-item'}
                    style={{ background: `url(${galleryPhoto3}) center center / cover `, height: '20rem' }}
                >
                    <div className={'article-info'}>
                        <h4>Food</h4>
                        <p>Lorem ipsum dolor sit amet.</p>
                        <a href="/">Read more</a>
                    </div>
                </Col>
            </Row>
            <Row>
                <Col
                    lg={12}
                    md={12}
                    sm={12}
                    style={{ background: `url(https://media1.giphy.com/media/l0ErNCJXTuHPQIUZq/giphy.gif?cid=790b7611fc6a5b2663ae7eec0a9d42b8cc9e97735e537279&rid=giphy.gif) center / contain `, height: '20rem' }}
                ></Col>
            </Row>
        </Container>
    );
};
