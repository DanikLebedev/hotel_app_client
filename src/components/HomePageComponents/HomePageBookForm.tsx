import React, { ChangeEvent, useCallback, useContext, useEffect, useState } from 'react';
import { Col, Container } from 'react-bootstrap';
import { Row } from 'react-bootstrap';
import { Category, Room, Rooms } from '../../interfaces/clientInterfaces';
import { useHttp } from '../../hooks/http.hook';
import { AuthContext } from '../../context/auth.context';
import { useHistory } from 'react-router-dom';
import { RoomService } from '../../APIServices/roomService';
import { CategoryService } from '../../APIServices/categoryService';
import toaster from 'toasted-notes';
import ScrollableAnchor from 'react-scrollable-anchor';

export const HomePageBookForm = () => {
    const [filteredRooms, setFilteredRooms] = useState<Room[]>([]);
    const [fetchedCategories, setFetchedCategories] = useState<Category[]>([]);
    const { error, clearError } = useHttp();
    const auth = useContext(AuthContext);
    const isAuthenticated = auth.isAuthenticated;
    const history = useHistory();
    const selectOrderChangeHandler = async (event: ChangeEvent<HTMLSelectElement>): Promise<void> => {
        event.persist();
        const { rooms }: Rooms = await RoomService.getAllRooms();
        const filteredRooms = rooms.filter(item => {
            return item.category.toString() === event.target.value.toString() && !item.isBooked;
        });
        setFilteredRooms(filteredRooms);
        // setShowedFreeRooms(filteredRooms);
        // setOrder({ ...order, category: event.target.value, price });
    };

    const fetchCategories = useCallback(() => {
        CategoryService.getAllCategories().then(({ categories }) => setFetchedCategories(categories));
    }, []);

    // const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
    //     setOrder({ ...order, [event.target.name]: event.target.value });
    // };

    const addOrderHandler = async () => {
        if (filteredRooms.length === 0) {
            toaster.notify('Sorry, all rooms are booked', {
                duration: 2000,
            });
        } else {
            history.push(`/rooms/${filteredRooms[0]._id}`);
        }

        // const data = await OrderService.postOrder(
        //     { ...order },
        //     {
        //         Authorization: `Bearer ${auth.token}`,
        //         'Content-Type': 'application/json',
        //     },
        // );
        // toaster.notify(data.message, {
        //     duration: 2000,
        // });
        // if (!isAuthenticated) {
        //     history.push('/auth');
        // }
    };

    const options = fetchedCategories.map(({ title }, index) => {
        return (
            <option key={title + index} value={title}>
                {title}
            </option>
        );
    });

    useEffect(() => {
        fetchCategories();
        toaster.notify(error, {
            duration: 2000,
        });
        clearError();
    }, [fetchCategories, error, clearError]);

    return (
        <Container>
            <ScrollableAnchor id={'home-page-book-form'}>
                <Row>
                    <Col lg={12} md={12} sm={12}>
                        <div className="d-flex home-page__book-form justify-content-center align-items-center flex-column">
                            <h2>Choose Your Room</h2>
                            <select
                                className={'form-control w-50 m-2'}
                                onChange={selectOrderChangeHandler}
                                name="category"
                            >
                                {options}
                            </select>
                            <button className={'button btn-black'} onClick={addOrderHandler}>
                                Find
                            </button>
                        </div>
                    </Col>
                </Row>
            </ScrollableAnchor>
        </Container>
    );
};
