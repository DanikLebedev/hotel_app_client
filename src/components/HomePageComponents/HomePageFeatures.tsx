import React from 'react';
import { Card, Col, Container, Row } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBreadSlice, faKey, faUserFriends } from '@fortawesome/free-solid-svg-icons';


export const HomePageFeatures: React.FC = (): JSX.Element => {
    return (
        <section className="home-page__features mb-5">
            <Container>
                <Row>
                    <Col lg={4}>
                        <Card className="border-0">
                            <Card.Body>
                                <FontAwesomeIcon size={'4x'} icon={faBreadSlice} />
                                <Card.Title>Free Breakfast Every Day</Card.Title>
                                <Card.Text>
                                    Every day in the morning we have a special buffe and it's free. Lorem ipsum dolor
                                    sit amet, consectetur adipisicing elit. Exercitationem, rerum vel. Doloribus
                                    exercitationem iusto laudantium.
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </Col>
                    <Col lg={4}>
                        <Card className="border-0">
                            <Card.Body>
                                <FontAwesomeIcon size={'4x'} icon={faUserFriends} />
                                <Card.Title>Attenive And Open Stuff</Card.Title>
                                <Card.Text>
                                    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci consequatur dicta
                                    dolore, ex, fuga hic in inventore modi nihil nisi officiis possimus qui quo
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </Col>
                    <Col lg={4}>
                        <Card className="border-0">
                            <Card.Body>
                                <FontAwesomeIcon size={'4x'} icon={faKey} />
                                <Card.Title>Luxurios Rooms</Card.Title>
                                <Card.Text>
                                    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquam aperiam corporis
                                    delectus deleniti distinctio, enim exercitationem perspiciatis porro provident,
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </Col>
                </Row>
            </Container>
        </section>
    );
};
