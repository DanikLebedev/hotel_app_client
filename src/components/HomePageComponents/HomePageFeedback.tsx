import React from 'react';
import { Carousel, Col, Container, Row } from 'react-bootstrap';

export const HomePageFeedback: React.FC = (): JSX.Element => {
    return (
        <section className="home-page__feedback mt-5 mb-5">
            <Container>
                <Row>
                    <Col lg={12} className="text-center">
                        <h2 className={'home-page__feedback-title section__title'}>Testimonials</h2>
                        <h4 className={'home-page__feedback-subtitle'}>what our happy cusomers said about us</h4>
                        <Carousel controls={false} className="p-2 home-page__feedback-slider">
                            <Carousel.Item>
                                <p>
                                    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi nisi quisquam
                                    sapiente. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi nisi
                                    quisquam sapiente. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi
                                    nisi quisquam sapiente.
                                </p>
                            </Carousel.Item>
                            <Carousel.Item>
                                <p>
                                    L12312321orem ipsum dolor sit amet, consectetur adipisicing elit. Commodi nisi
                                    quisquam sapiente. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi
                                    nisi quisquam sapiente. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                                    Commodi nisi quisquam sapiente.
                                </p>
                            </Carousel.Item>
                            <Carousel.Item>
                                <p>
                                    12312Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi nisi quisquam
                                    sapiente. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi nisi
                                    quisquam sapiente. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi
                                    nisi quisquam sapiente.
                                </p>
                            </Carousel.Item>
                        </Carousel>
                    </Col>
                </Row>
            </Container>
        </section>
    );
};
