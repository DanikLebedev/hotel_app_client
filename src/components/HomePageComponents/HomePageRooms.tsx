import React, { useCallback, useContext, useEffect, useState } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBed, faBuilding, faUserAlt } from '@fortawesome/free-solid-svg-icons';
import { NavLink, useHistory } from 'react-router-dom';
import { Room } from '../../interfaces/clientInterfaces';
import { RoomService } from '../../APIServices/roomService';
import Loader from '../Loader/Loader';
import { config } from '../../config';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import 'react-lazy-load-image-component/src/effects/opacity.css';
import { withTranslation } from 'react-i18next';
import { ClientContext } from '../../context/client.context';

const HomePageRooms: React.FC = ({ t }: any): JSX.Element => {
    const [index, setIndex] = useState<number>(0);
    const history = useHistory();
    const fetchedRooms = useContext(ClientContext).fetchedRooms;


    const changeMainRoomHandler = (key: number): void => {
        setIndex(key);
    };


    return (
        <section className={'home__page-rooms'}>
            <Container fluid={true}>
                <Row className="home__page-rooms-preview">
                    <Col lg={6} md={6} sm={12} xs={12} className="p-0 home__page-rooms-preview-main-photo">
                        {fetchedRooms[index] ? (
                            <LazyLoadImage
                                effect="opacity"
                                src={config.baseUrl + fetchedRooms[index].image}
                                alt="room"
                            />
                        ) : (
                            <Loader />
                        )}
                    </Col>
                    <Col lg={6} md={6} sm={12} xs={12}>
                        <h2 className={'room__title section__title'}>
                            {fetchedRooms[index] ? fetchedRooms[index].title : null}
                        </h2>
                        <div className={'room-info'}>
                            <p>
                                <span className={'home__page-rooms-price'}>
                                    {fetchedRooms[index] ? fetchedRooms[index].price : null}$
                                </span>
                                /{t('home-page-rooms.per-night.label')}
                            </p>
                        </div>
                        <div className="room__icons">
                            <span>
                                <FontAwesomeIcon className={'mr-2'} icon={faBed} />
                                {fetchedRooms[index] ? fetchedRooms[index].rooms : null}{' '}
                                {t('home-page-rooms.rooms.label')}
                            </span>
                            <span>
                                <FontAwesomeIcon className={'mr-2'} icon={faUserAlt} />
                                {fetchedRooms[index] ? fetchedRooms[index].guests : null}{' '}
                                {t('home-page-rooms.guests.label')}
                            </span>
                            <span>
                                <FontAwesomeIcon className={'mr-2'} icon={faBuilding} />
                                {fetchedRooms[index] ? fetchedRooms[index].area : null}m<sup>2</sup>
                            </span>
                        </div>
                        <p className={'home__page-rooms-description'}>
                            {fetchedRooms[index]
                                ? t(`home-page-rooms.${fetchedRooms[index].category}.description`)
                                : null}
                        </p>
                        <div className={'button__container'}>
                            <button
                                id={fetchedRooms[index] ? fetchedRooms[index]._id : ''}
                                onClick={() => history.push(`/rooms/${fetchedRooms[index]._id}`)}
                                className={'button btn-black'}
                            >
                                Book Room
                            </button>
                            <NavLink to={'/rooms'}>
                                <button className={'button btn-black'}>See Room</button>
                            </NavLink>
                        </div>
                    </Col>
                </Row>
                <div className="justify-content-center home-page-rooms-row mt-3 d-flex align-items-center">
                    {fetchedRooms ? (
                        fetchedRooms.slice(0, 4).map((item, key) => {
                            return (
                                <div
                                    onClick={() => changeMainRoomHandler(key)}
                                    key={key}
                                    className="home__page-rooms-item col-lg-3 col-md-6 col-sm-12 pl-0 mb-2"
                                >
                                    <LazyLoadImage effect="opacity" src={config.baseUrl + item.image} alt="room" />
                                    <div className={'home__page-rooms-item-title'}>
                                        <span>{item.title}</span>
                                        <span>{item.price}$</span>
                                    </div>
                                </div>
                            );
                        })
                    ) : (
                        <Loader />
                    )}
                </div>
            </Container>
        </section>
    );
};

export default withTranslation()(HomePageRooms);
