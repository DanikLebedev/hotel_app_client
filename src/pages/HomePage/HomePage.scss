@import "src/index";
@import "src/components/Navigation/Navigation";



.main__title {
  margin-bottom: 15vh;
  margin-top: 15vh;
}

.room__icons {
  display: flex;
  flex-direction: column;
  span {

  }
}


.home-page__book-form {
  padding: 10px;
  border-radius: 20px;
  transition: .3s ease-in-out;
  margin: 10px auto;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  width: 50%;
  box-shadow: 0 0 10px rgba(0,0,0,0.5);
  .button {
    margin-top: 10px;
    background: $price-color;
  }
}

.home-page__header {
  padding-bottom: 180px;
}

.header__title {
  font-weight: bold;
  color: #ffffff;
  font-size: 4rem;
  text-align: center;
}

.header__subtitle {
  font-size: 1.5em;
  color: $base-text-color;
  max-width: 40vw;
}

.button__container {
  display: flex;
}

.home-page__about-us {
  padding: 15vh 0 10vh 0;

  &-photos {
    img {
      height: 20rem;
      max-width: 300px;
      animation: slideInRight .8s ease-in-out;
    }
  }

  &-text {
    max-width: 80%;
    font-size: 1.3em;
  }
}


@keyframes fly {
  from {
    -webkit-transform: translateY(0);
    transform: translateY(0);
  }

  50% {
    -webkit-transform: translateY(-20px);
    transform: translateY(-20px);
  }

  to {
    -webkit-transform: translateY(0);
    transform: translateY(0);
  }

}

.arrow-icon {
  font-size: 5em;
  margin-top: 5rem;
  animation: fly 2s ease-in-out infinite;
}


.button {
  padding: 8px 20px;
  margin-right: 10px;
  background-color: $base-text-color;
  color: $base-dark-button-color;
  transition: .3s ease;
  font-size: 1.4em;
  border: 1px solid #ffffff;
  white-space: nowrap;
  border-radius: 3px;

  &:hover {
    background: #000000;
    color: $base-text-color;
  }
}

.btn-transparent {
  background: transparent;
  color: #ffffff;
  border: 1px solid #ffffff;
}

.btn-black {
  background: #000000;
  color: #ffffff;

  &:hover {
    background: #ffffff;
    color: #000000;
    border-color: #000;
  }
}

.section__title {
  font-weight: bold;
  font-size: 3em;
}

.home__page-rooms {
  padding-bottom: 30px;
  &-price {
    font-size: 2em;
    color: $price-color;
    font-weight: bold;
  }

  &-preview {
    transition: .5s ease;

    img {
      width: 90%;
      height: 25rem;
      object-fit: cover;
    }
  }

  &-photo {
    width: 45vw
  }



  &-item {
    position: relative;
    height: 20vh;
    margin-right: 10px;
    transition: .3s ease;

    img {
      position: absolute;
      width: 100%;
      height: 100%;
      object-fit: cover;
    }

    &:hover {
      transform: translateY(-3px);
      cursor: pointer;
    }

    &:last-child {
      margin-right: 0;
    }

    &-title {
      display: flex;
      margin-top: 10px;
      font-weight: bold;
      position: absolute;
      top: 7%;
      left: 3%;

      span {
        display: block;
        padding: 10px;
        background: $base-dark-button-color;
        color: $base-text-color;
      }

      span:nth-child(2) {
        color: $price-color;
      }
    }

  }

}

.home-page__feedback {
  &-slider {
    position: relative;

    &:before {
      content: '\201C';
      position: absolute;
      font-size: 40px;
      top: 0;
      left: 0;
      color: $price-color;
    }

    &:after {
      content: '\201D';
      font-size: 40px;
      position: absolute;
      bottom: 0;
      right: 0;
      color: $price-color;
    }

    p {
      font-size: 1.5em;
      max-width: 100%;
      margin-left: 16px;
    }
  }
}

@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 1) {

  .home-page__header {
    padding-bottom: 140px;
  }

  .header__title {
    font-size: 3.1em;
  }

  .home-page__feedback {
    p {
      font-size: 1.3em;
    }
  }

  .header__subtitle {
    font-size: 1.3em;
  }
  .arrow-icon {
    margin-top: 4.5rem;
  }

  .home-page__about-us {
    padding-bottom: 5vh;

    &-title {
      font-size: 2.5em;
    }

    img {

    }

    &-text {
      font-size: 1em;
    }
  }

}

@media screen and (min-width: 768px) and (max-width: 1199px) {
  .home-page-rooms-row {
    flex-wrap: wrap;
  }
  .home-page__header {
    padding-bottom: 150px;
  }

  .home-page__feedback {
    &-title {
      font-size: 1.3em;
    }

    &-subtitle {
      font-size: 1.2em;
    }

    p {
      font-size: 1.1em;
    }
  }

  .header__title {
    font-size: 2.7em;
  }


  .header__subtitle {
    font-size: 1.1em;
  }

  .arrow-icon {
    margin-top: 3.5rem;
    font-size: 3.5rem;
  }

  .home-page__about-us {
    padding-bottom: 5vh;

    &-title {
      font-size: 2em;
    }

    img {
      height: 15rem;
    }

    &-text {
      font-size: 1em;
    }
  }
}


.photo-wrapper {
  background: url("../../assets/images/38736452-luxurious-bed-with-cushion-in-royal-bedroom-interior-hotel-room-in-bright-colors-vertical-orientatio.jpg") center center / cover;
}

@media screen and (max-width: 768px) and (min-width: 480px) {
  .home-page-rooms-row {
    flex-wrap: wrap;
  }

  .button {
    font-size: 1.2em;
  }

  .home-page__feedback {
    &-title {
      font-size: 1.3em;
    }

    &-subtitle {
      font-size: 1.2em;
    }

    p {
      font-size: 1.1em;
    }
  }

  .home-page__header {
    padding-bottom: 80px;
  }

  .header__title {
    font-size: 2.7em;
  }


  .header__subtitle {
    font-size: 1.1em;
  }

  .arrow-icon {
    margin-top: 3.5rem;
    font-size: 3.5rem;
  }

  .home-page__about-us {
    &-photos {
      margin-top: 30px;
    }

    &-title {
      font-size: 2em;
    }

    img {
      height: 15rem;
    }

    &-text {
      font-size: 1em;
    }
  }

  .home__page-rooms-item-title {
    span {
      font-size: 0.8em;
    }
  }

  .room__title {
    font-size: 1.5em;
  }
  .social_icon {
    width: 20px;
    height: 20px;
  }

}


@media screen and (min-width: 300px) and (max-width: 480px) {

  .grid {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
  }

  .arrow-icon {
    font-size: 2em;
    margin-top: 3rem;
  }
  .home__page-rooms-preview  {
    img {
      height: 15rem;
    }
  }
  .separator {
    display: none;
  }

  .home-page__header {
    padding-bottom: 60px;
  }


  .header__title {
    font-size: 2em;
  }

  .header__subtitle {
    font-size: 1em;
    max-width: 80%;
  }

  .button {
    font-size: 1em;
  }


  .navbar-expand {
    justify-content: center !important;
  }

  .settings-wrapper {
    flex-direction: column;
    align-items: flex-end;
  }
  ul {
    .social_icon {
      width: 30px;
      height: 30px;
    }
  }

  .home-page-rooms-row {
    flex-wrap: wrap;
  }

  .home-page__about-us {
    padding-top: 0;
    &-title {
      font-size: 1.2em;
    }

    &-text {
      font-size: 1em;
    }
  }

  .room__title {
    font-size: 1.2em;
  }

  .room__icons {
    font-size: 1em;
  }

  .home__page-rooms-item-title {
    font-size: 0.8em;
  }

  .home-page__feedback {
    &-title {
      font-size: 1.5em;
    }
    &-subtitle {
      font-size: 1.2em;
    }
    &-slider {
      p {
        font-size: 1em;
      }
    }
  }

  .home-page__book-form {
    h2 {
      font-size: 1.2em;
    }
  }

  .footer-contacts-wrapper {
    flex-direction: row !important;
    margin: 10px 0 10px 0;
    span {
      margin-right: 5px;
    }
  }

}


.home-page-articles {
  &-item {
    position: relative;
    overflow: hidden;
    transition: .5s ease-in-out;
    .article-info {
      background: #ffffff;
      width: 80%;
      padding: 10px;
      position: absolute;
      margin-left: 10px;
      bottom: 5%;
      left: 0;
      transform: translateY(140px);
      transition: .5s ease-in-out;
    }
    &:hover {
      filter: grayscale(.75);
      .article-info {
       transform: translateY(0);

      }
    }
  }
}