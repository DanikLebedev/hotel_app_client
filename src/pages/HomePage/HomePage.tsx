import React from 'react';
import './HomePage.scss';
import HomePageAboutUs from '../../components/HomePageComponents/HomePageAboutUs';
import HomePageRooms from '../../components/HomePageComponents/HomePageRooms';
import HomePageFeedback from '../../components/HomePageComponents/HomePageFeedback';
import HomePageFeatures from '../../components/HomePageComponents/HomePageFeatures';
import { LazyLoadComponent } from 'react-lazy-load-image-component';
import { HomePageArticles } from '../../components/HomePageComponents/HomePageArticles';
import FindRoomForm from '../../components/FindRoomForm/FindRoomForm';
import { MapComponent } from '../../components/YandexMap/YandexMap';
import ErrorBoundary from '../../components/ErrorsComponents/ErrorBoundary';

export const HomePage: React.FC = () => {
    return (
        <>
            <FindRoomForm />
            <HomePageAboutUs />
            <ErrorBoundary>
                <LazyLoadComponent>
                    <HomePageRooms />
                </LazyLoadComponent>
            </ErrorBoundary>
            <ErrorBoundary>
                <HomePageFeedback />
            </ErrorBoundary>
            <HomePageFeatures />
            <ErrorBoundary>
                <HomePageArticles />
            </ErrorBoundary>
            <MapComponent />
        </>
    );
};
