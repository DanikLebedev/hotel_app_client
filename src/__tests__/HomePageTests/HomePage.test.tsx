import React from 'react';
import { HomePage } from '../../pages/HomePage/HomePage';
import { shallow } from 'enzyme';
import HomePageAboutUs from '../../components/HomePageComponents/HomePageAboutUs';
import HomePageRooms from '../../components/HomePageComponents/HomePageRooms';
import HomePageFeatures from '../../components/HomePageComponents/HomePageFeatures';
import HomePageFeedback from '../../components/HomePageComponents/HomePageFeedback';
import { HomePageArticles } from '../../components/HomePageComponents/HomePageArticles';

it('should renders without crashing', function() {
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toMatchSnapshot();
});

describe('render home page components', () => {
    it('should renders without crashing', function() {
        const wrapper = shallow(<HomePageAboutUs />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should renders without crashing', function() {
        const wrapper = shallow(<HomePageRooms />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should renders without crashing', function() {
        const wrapper = shallow(<HomePageFeatures />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should renders without crashing', function() {
        const wrapper = shallow(<HomePageFeedback />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should renders without crashing', function() {
        const wrapper = shallow(<HomePageArticles />);
        expect(wrapper).toMatchSnapshot();
    });
});
