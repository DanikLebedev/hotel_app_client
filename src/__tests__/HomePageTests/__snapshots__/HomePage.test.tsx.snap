// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home page components should renders without crashing 1`] = `
<HomePageAboutUs
  i18n={Object {}}
  t={[Function]}
  tReady={false}
/>
`;

exports[`render home page components should renders without crashing 2`] = `
<HomePageRooms
  i18n={Object {}}
  t={[Function]}
  tReady={false}
/>
`;

exports[`render home page components should renders without crashing 3`] = `
<HomePageFeatures
  i18n={Object {}}
  t={[Function]}
  tReady={false}
/>
`;

exports[`render home page components should renders without crashing 4`] = `
<HomePageFeedback
  i18n={Object {}}
  t={[Function]}
  tReady={false}
/>
`;

exports[`render home page components should renders without crashing 5`] = `
<Container
  fluid={true}
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      className="home-page-articles-item"
      lg={4}
      md={4}
      sm={4}
      style={
        Object {
          "background": "url(cream_candles_flowers_green_therapy_11044_1600x1200.jpg) center center / cover",
          "height": "20rem",
        }
      }
    >
      <div
        className="article-info"
      >
        <h4>
          Health SPA
        </h4>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
        <a
          href="/"
        >
          Read more
        </a>
      </div>
    </Col>
    <Col
      className="home-page-articles-item"
      lg={4}
      md={4}
      sm={4}
      style={
        Object {
          "background": "url(thumb-1920-206717.jpg) center center / cover",
          "height": "20rem",
        }
      }
    >
      <div
        className="article-info"
      >
        <h4>
          Golf
        </h4>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
        <a
          href="/"
        >
          Read more
        </a>
      </div>
    </Col>
    <Col
      className="home-page-articles-item"
      lg={4}
      md={4}
      sm={4}
      style={
        Object {
          "background": "url(10-103370_salad-food-wallpaper-food-images-high-resolution.jpg) center center / cover ",
          "height": "20rem",
        }
      }
    >
      <div
        className="article-info"
      >
        <h4>
          Food
        </h4>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
        <a
          href="/"
        >
          Read more
        </a>
      </div>
    </Col>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <Col
      lg={12}
      md={12}
      sm={12}
      style={
        Object {
          "background": "url(colombo_2.jpg) center center / cover ",
          "height": "20rem",
        }
      }
    />
  </ForwardRef>
</Container>
`;

exports[`should renders without crashing 1`] = `
<Fragment>
  <withI18nextTranslation(FindRoomForm) />
  <withI18nextTranslation(HomePageAboutUs) />
  <t
    afterLoad={[Function]}
    beforeLoad={[Function]}
    useIntersectionObserver={true}
    visibleByDefault={false}
  >
    <withI18nextTranslation(HomePageRooms) />
  </t>
  <withI18nextTranslation(HomePageFeedback) />
  <withI18nextTranslation(HomePageFeatures) />
  <HomePageArticles />
  <MapComponent />
</Fragment>
`;
